sui/src/benchmark/validator_preparer.rs

```diff
@@ -78,16 +82,17 @@ impl ValidatorPreparer {
         let (key_pairs, mut genesis_config) = set_up_authorities_and_committee(committee_size)
             .expect("Got error in setting up committee");
 
-        // pick the first keypair as main validator
-        let main_authority_address_hex =
-            format!("{}", SuiAddress::from(key_pairs[0].public_key_bytes()));
+        let main_authority_address_hex = format!(
+            "{}",
+            SuiAddress::from(genesis_config.key_pair.public_key_bytes())
+        );
         info!("authority address hex: {}", main_authority_address_hex);
 
-        let mut keys = Vec::new();
-        for key_pair in key_pairs {
-            let name = *key_pair.public_key_bytes();
-            keys.push((name, key_pair));
-        }
+        let keys: Vec<_> = key_pairs
+            .into_iter()
+            .map(|key_pair| (*key_pair.public_key_bytes(), key_pair))
+            .collect();
+
         let committee = Committee::new(0, keys.iter().map(|(k, _)| (*k, 1)).collect());
```

Comment:
the `key_pair` is also used afterwards, e.g. line 105, line 133 etc

Reply:
Needlessly so. You should never need to couple the `KeyPair` with a `PublicKey`, since for any `kp: KeyPair`, `kp.public_key_bytes()` will give you the public key, and this computation is memoized (you'll never pay for it more than once).

I opened https://github.com/longbowlu/sui/pull/1 to fix.